homeassistant/components/prana/coordinator.py

```diff
@@ -0,0 +1,58 @@
+"""Coordinator for Prana integration.
+
+Responsible for polling the device REST endpoints and normalizing data for entities.
+"""
+
+from datetime import timedelta
+import logging
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from . import PranaConfigEntry
+
+
+from prana_local_api_client.exceptions import (
+    PranaApiCommunicationError,
+    PranaApiUpdateFailed,
+)
+from prana_local_api_client.models.prana_state import PranaState
+from prana_local_api_client.prana_api_client import PranaLocalApiClient
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_HOST, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class PranaCoordinator(DataUpdateCoordinator[PranaState]):
+    """Universal coordinator for Prana (fan, switch, sensor, light data)."""
+
+    def __init__(self, hass: HomeAssistant, entry: "PranaConfigEntry") -> None:
```

Comment:
```suggestion
    def __init__(self, hass: HomeAssistant, entry: PranaConfigEntry) -> None:
```
move the `type` to this file if you are having issues with imports